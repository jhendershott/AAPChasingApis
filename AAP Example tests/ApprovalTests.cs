using AAPExampleTests;
using ApprovalTests;
using ApprovalTests.Reporters;
using RestSharp;
using System;
using System.Collections.Generic;
using System.Text;
using System.Text.RegularExpressions;
using Xunit;

namespace AAP_Example_tests
{   
    [UseReporter(typeof(DiffReporter))]
    public class ApprovalTests
    {
        [Fact]
        public void ApprovalTestswithVersionCompare()
        {
            var actual = Request.RestRequest("https://api.exchangeratesapi.io/", "2010-01-14", "GET");
            Approvals.Verify(ScrubDynamic(actual.Content));
        }

        [Fact]
        public void ApprovalTestswithVersionCompareDynamic()
        {
            var actual = Request.RestRequest("https://api.exchangeratesapi.io/", "2010-01-14", "GET");
            Approvals.Verify(ScrubDynamic(actual.Content));
        }


        private string ScrubDynamic(string resp)
        {
            string scrubbedResponse = Regex.Replace(resp, "\"date\":\"(.*?)\"", "###DATE");
            return scrubbedResponse;
        }
    }
}
