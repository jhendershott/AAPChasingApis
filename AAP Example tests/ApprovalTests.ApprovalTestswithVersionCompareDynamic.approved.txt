{"rates":{"CAD":1.4752,"HKD":11.1536,"LVL":0.7085,"PHP":65.867,"DKK":7.442,"HUF":266.8,"CZK":25.947,"AUD":1.5547,"RON":4.1043,"SEK":10.143,"IDR":13245.61,"INR":65.7,"BRL":2.5438,"RUB":42.5015,"LTL":3.4528,"JPY":130.76,"THB":47.244,"CHF":1.4756,"SGD":1.9984,"PLN":4.0399,"BGN":1.9558,"TRY":2.09,"CNY":9.813,"NOK":8.143,"NZD":1.9497,"ZAR":10.6588,"USD":1.4374,"MXN":18.279,"EEK":15.6466,"GBP":0.88105,"KRW":1614.29,"MYR":4.8002,"HRK":7.289},"base":"EUR",###DATE}